models/user.rb

```diff
@@ -0,0 +1,20 @@
+class User < ActiveRecord::Base
+  validates :point, :numericality => { :greater_than_or_equal_to => 0 }
+
+  # speeches whose speaker is this user
+  has_many :speeches
+
+  # association between speeches and users
+  has_many :audiences
+  # speeches this user has applied as an audience
+  has_many :joined_speeches, :through => :audiences, :source => :speech
```

Comment:
The intuitive understanding for the name `joined_speeches` are -- "The speeches he joined both as speakers and audiences"